### PR TITLE
rest-plugin

### PR DESCRIPTION
**Pull request by kosteman**

### CODE REVIEW

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                        || annotation instanceof Query && ((Query) annotation).name().equals(name)
+                        || annotation instanceof Header && ((Header) annotation).name().equals(name)
+                        && value != null && !value.isEmpty()) {
+                    set(endpoint, field, value);
+                    return;
+                }
+            }
+        }
+
+        throw new RestPluginException("There is no '" + name + "' parameter in '" + endpoint.getClass().getAnnotation(ru.sbtqa.tag.api.annotation.Endpoint.class).title() + "' endpoint.");
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: 1. Exteract variable of endpoint.getClass().getAnnotation(ru.sbtqa.tag.api.annotation.Endpoint.class).title()
2. use String.format

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    public String getBody(String template) {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: Add javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            } else {
+                LOG.debug("Failed to substitute not String field to body template");
+            }
+        }
+
+        return body;
+    }
+
+
+    public void validate(String title, Object... params) {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: add javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        Method[] methods = endpoint.getClass().getMethods();
+        for (Method method : methods) {
+            if (null != method.getAnnotation(Validation.class)
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: extract variable

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                try {
+                    method.invoke(endpoint, params);
+                } catch (InvocationTargetException | IllegalAccessException | IllegalArgumentException e) {
+                    throw new RestPluginException("Failed to invoke method", e);
+                }
+                return;
+            }
+        }
+
+        throw new RestPluginException("There is no '" + title + "' validation rule in '" + endpoint.getClass().getAnnotation(ru.sbtqa.tag.api.annotation.Endpoint.class).title() + "' endpoint.");
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: String.format + variable

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+    public void validate(String title, Object... params) {
+        Method[] methods = endpoint.getClass().getMethods();
+        for (Method method : methods) {
+            if (null != method.getAnnotation(Validation.class)
+                    && method.getAnnotation(Validation.class).title().equals(title)) {
+                try {
+                    method.invoke(endpoint, params);
+                } catch (InvocationTargetException | IllegalAccessException | IllegalArgumentException e) {
+                    throw new RestPluginException("Failed to invoke method", e);
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: Throw more detailed exception like "Failed to validate..."

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java (hunk 5 above)
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: String.format + variable

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+                }
+                return;
+            }
+        }
+
+        throw new RestPluginException("There is no '" + title + "' validation rule in '" + endpoint.getClass().getAnnotation(ru.sbtqa.tag.api.annotation.Endpoint.class).title() + "' endpoint.");
+    }
+
+    public Map<String, Object> getParameters(ParameterType type) {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/HTTP.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+public enum HTTP {
+    POST,
+    GET,
+    PUT,
+    DELETE,
+    PATCH,
+    OPTIONS,
+    HEAD;
+
+    public static boolean isBodiless(HTTP http) {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/HTTP.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/Body.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package ru.sbtqa.tag.api.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
+public @interface Body {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/Body.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/Body.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+    String name();
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/Body.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/Endpoint.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import ru.sbtqa.tag.api.HTTP;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface Endpoint {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/Endpoint.java]: javadoc for annotation and all content

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/FromResponse.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Dependent response. Optional annotation. It will be used to try to initialize
+ * endpoint parameter from response of previous executed request to api.
+ *
+ *
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
+public @interface FromResponse {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/FromResponse.java]: Javadoc for all content

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/Header.java [hunk 13]
```diff
@@ -0,0 +1,13 @@
+package ru.sbtqa.tag.api.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
+public @interface Header {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/Header.java]: javadoc for all content

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/Query.java [hunk 14]
```diff
@@ -0,0 +1,13 @@
+package ru.sbtqa.tag.api.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
+public @interface Query {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/Query.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/Applicator.java [hunk 15]
```diff
@@ -0,0 +1,6 @@
+package ru.sbtqa.tag.api.annotation.applicators;
+
+public interface Applicator {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/Applicator.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/ApplicatorHandler.java [hunk 16]
```diff
@@ -0,0 +1,46 @@
+package ru.sbtqa.tag.api.annotation.applicators;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import ru.sbtqa.tag.api.exception.RestPluginException;
+
+public class ApplicatorHandler<T extends Applicator> {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/ApplicatorHandler.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/ApplicatorHandler.java [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+
+    private List<T> applicators = new ArrayList<>();
+
+    public void add(T applicator) {
+        applicators.add(applicator);
+    }
+
+    public void apply() {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/ApplicatorHandler.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/ApplicatorHandler.java [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+            applicator.apply();
+        }
+
+        applicators.clear();
+    }
+
+    private List<T> sort() {
+        return applicators.stream()
+                .sorted(Comparator.comparing(applicator -> {
+                    if (applicator.getClass().getAnnotation(Order.class) != null) {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/ApplicatorHandler.java]: extract variable

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/DefaultApplicator.java [hunk 19]
```diff
@@ -0,0 +1,24 @@
+package ru.sbtqa.tag.api.annotation.applicators;
+
+import java.lang.reflect.Field;
+import ru.sbtqa.tag.api.EndpointEntry;
+import ru.sbtqa.tag.api.utils.ReflectionUtils;
+
+public abstract class DefaultApplicator {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/DefaultApplicator.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/FromResponseApplicator.java [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+import java.lang.reflect.Field;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import ru.sbtqa.tag.api.EndpointEntry;
+import ru.sbtqa.tag.api.environment.ApiEnvironment;
+import ru.sbtqa.tag.api.annotation.FromResponse;
+import ru.sbtqa.tag.api.exception.RestPluginException;
+
+@Order(0)
+public class FromResponseApplicator extends DefaultApplicator implements Applicator {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/FromResponseApplicator.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/manager/EndpointManager.java [hunk 21]
```diff
@@ -0,0 +1,67 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.sbtqa.tag.api.EndpointEntry;
+import ru.sbtqa.tag.api.annotation.Endpoint;
+import ru.sbtqa.tag.api.context.EndpointContext;
+import ru.sbtqa.tag.api.exception.RestPluginException;
+import ru.sbtqa.tag.api.properties.ApiConfiguration;
+import ru.sbtqa.tag.qautils.errors.AutotestError;
+
+public class EndpointManager {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/manager/EndpointManager.java]: javadoc

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ApiGenericStepDefs.java [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+ */
+public class ApiGenericStepDefs extends ApiSetupSteps {
+
+    /**
+     * Execute endpoint action (request) with no parameters
+     *
+     * @param action name value of the endpoint annotation to execute
+     * @throws RestPluginException if there is an error while endpoint executing
+     */
+    public void userSendRequestNoParams(String action) {
```
Comment (clicman) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ApiGenericStepDefs.java]: Set all stepdefs to be a static

File: plugins/rest-plugin/src/test/resources/templates/Client.json [hunk 23]
```diff
@@ -0,0 +1,5 @@
+{
+  "id": %id,
```
Comment (clicman) [inline plugins/rest-plugin/src/test/resources/templates/Client.json]: We'we decided to do smth with % placeholders

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java (hunk 7 above)
Comment (sbtqasonar) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: Code Smell Code Smell: Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2KVJuW2AOP0CkD5Fx&open=AWU2KVJuW2AOP0CkD5Fx)

[ISSUE_KEY]: AWU2KVJuW2AOP0CkD5Fx

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/utils/ReflectionUtils.java [hunk 24]
```diff
@@ -0,0 +1,27 @@
+package ru.sbtqa.tag.api.utils;
+
+import java.lang.reflect.Field;
+import ru.sbtqa.tag.api.EndpointEntry;
+import ru.sbtqa.tag.api.exception.RestPluginException;
+
+public class ReflectionUtils {
```
Comment (sbtqasonar) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/utils/ReflectionUtils.java]: Code Smell Code Smell: Add a private constructor to hide the implicit public one. ([squid:S1118](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1118))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2KVKnW2AOP0CkD5Fz&open=AWU2KVKnW2AOP0CkD5Fz)

[ISSUE_KEY]: AWU2KVKnW2AOP0CkD5Fz

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java [hunk 25]
```diff
@@ -0,0 +1,161 @@
...
+            field.setAccessible(true);
+
+            ApplicatorHandler<Applicator> applicators = new ApplicatorHandler<>();
+            for (Annotation annotation : field.getAnnotations()) {
+                if (annotation instanceof FromResponse) {
+                    applicators.add(new FromResponseApplicator(endpoint, field));
+                } else if (annotation instanceof Stashed) {
+                    applicators.add(new StashedApplicator(endpoint, field));
+                } else {
+                    continue;
```
Comment (sbtqasonar) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: Code Smell Code Smell: Remove this redundant jump. ([squid:S3626](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS3626))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2KVJuW2AOP0CkD5Fw&open=AWU2KVJuW2AOP0CkD5Fw)

[ISSUE_KEY]: AWU2KVJuW2AOP0CkD5Fw

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/manager/EndpointManager.java (hunk 21 above)
Comment (sbtqasonar) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/manager/EndpointManager.java]: Code Smell Code Smell: Add a private constructor to hide the implicit public one. ([squid:S1118](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1118))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2KVKaW2AOP0CkD5Fy&open=AWU2KVKaW2AOP0CkD5Fy)

[ISSUE_KEY]: AWU2KVKaW2AOP0CkD5Fy

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java [hunk 26]
```diff
@@ -0,0 +1,159 @@
...
+                    throw new RestPluginException("Failed to invoke method", e);
+                }
+                return;
+            }
+        }
+
+        throw new RestPluginException("There is no '" + title + "' validation rule in '" + endpoint.getClass().getAnnotation(ru.sbtqa.tag.api.annotation.Endpoint.class).title() + "' endpoint.");
+    }
+
+    public Map<String, Object> getParameters(ParameterType type) {
```
Comment (sbtqasonar) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: Code Smell Code Smell: Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2KVJuW2AOP0CkD5Fx&open=AWU2KVJuW2AOP0CkD5Fx)

[ISSUE_KEY]: AWU2KVJuW2AOP0CkD5Fx

File: page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/PageManager.java [hunk 27]
```diff
@@ -181,13 +145,13 @@ public static Page changeUrlByTitle(String title) throws PageInitializationExcep
      */
     public static Page changeUrlByTitle(String packageName, String title) throws PageInitializationException {
```
Comment (sbtqasonar) [inline page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/PageManager.java]: Code Smell Code Smell: Remove this unused method parameter "packageName". ([squid:S1172](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2kh_5W2AOP0CkD9zl&open=AWU2kh_5W2AOP0CkD9zl)

[ISSUE_KEY]: AWU2kh_5W2AOP0CkD9zl

Comment (sbtqasonar) [inline page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/PageManager.java]: Code Smell Code Smell: Remove this unused method parameter "packageName". ([squid:S1172](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2kh_5W2AOP0CkD9zl&open=AWU2kh_5W2AOP0CkD9zl)

[ISSUE_KEY]: AWU2kh_5W2AOP0CkD9zl

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java [hunk 28]
```diff
@@ -0,0 +1,192 @@
...
+        throw new RestPluginException(String.format("There is no \"%s\" validation rule in \"%s\" endpoint", title, entryTitle));
+    }
+
+    /**
+     * Get name-field map with fields annotated with one of {@link ParameterType} annotation
+     * @param type type of parameter
+     * @return name-field map
+     */
+
+    public Map<String, Object> getParameters(ParameterType type) {
```
Comment (sbtqasonar) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: Code Smell Code Smell: Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2KVJuW2AOP0CkD5Fx&open=AWU2KVJuW2AOP0CkD5Fx)

[ISSUE_KEY]: AWU2KVJuW2AOP0CkD5Fx

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/manager/EndpointManager.java (hunk 21 above)
Comment (kosteman) [reply plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/manager/EndpointManager.java]: излишне

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/DefaultApplicator.java (hunk 19 above)
Comment (kosteman) [reply plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/annotation/applicators/DefaultApplicator.java]: излишне

File: plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java (hunk 28 above)
Comment (sbtqasonar) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: Code Smell Code Smell: Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2KVJuW2AOP0CkD5Fx&open=AWU2KVJuW2AOP0CkD5Fx)

[ISSUE_KEY]: AWU2KVJuW2AOP0CkD5Fx

Comment (sbtqasonar) [inline plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java]: Code Smell Code Smell: Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2KVJuW2AOP0CkD5Fx&open=AWU2KVJuW2AOP0CkD5Fx)

[ISSUE_KEY]: AWU2KVJuW2AOP0CkD5Fx

**Pull request merged**